### PR TITLE
[FIX] Preprocess: pickle when saving

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Preprocess widget is unable to save _Randomize_, _Normalize Features_, and _Continuize Discrete Variables_ due to some pickle error.

##### Steps to reproduce the behavior
Add  _Randomize_ or _Normalize Features_ or _Continuize Discrete Variables_. Change some parameters. Save the workflow and close the application Orange. Then run the app and open the workflow. 

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by kernc** [comment]
How can this issue be reproduced?

**Comment by jerneju** [comment]
@kernc : Added the steps above.

File: Orange/preprocess/preprocess.py [hunk 1]
```diff
@@ -34,12 +34,20 @@ def __call__(self, data):
 
 
 class Continuize(Preprocess):
-    (Indicators, FirstAsBase, FrequentAsBase, Remove, RemoveMultinomial,
-     ReportError, AsOrdinal, AsNormalizedOrdinal, Leave) = Enum(
-         "Continuize",
-         "Indicators, FirstAsBase, FrequentAsBase,"
-         "Remove, RemoveMultinomial, ReportError, AsOrdinal,"
-         "AsNormalizedOrdinal, Leave")
+    class ContinuizeDV(Enum):
```
Comment (kernc) [inline Orange/preprocess/preprocess.py]: Before, you'd get:
```py
>>> from Orange.preprocess import Continuize
>>> preproc = Continuize(multinomial_treatment=Continuize.FrequentAsBase)
>>> preproc
Continuize(multinomial_treatment=Continuize.FrequentAsBase)
```
With this change, it's:
```py
>>> preproc = Continuize(multinomial_treatment=Continuize.FrequentAsBase)
>>> preproc
Continuize(multinomial_treatment=2)
```
Is there some way to keep the customized Enum and still pickle numbers?

Or, as an alternative, sklearn uses expressive strings. I wouldn't that also.

Comment (jerneju) [inline Orange/preprocess/preprocess.py]: @kernc: I did some changes.

File: Orange/preprocess/preprocess.py [hunk 2]
```diff
@@ -34,12 +34,20 @@ def __call__(self, data):
...
+        Indicators = "Indicators"
+        FirstAsBase = "FirstAsBase"
+        FrequentAsBase = "FrequentAsBase"
+        Remove = "Remove"
+        RemoveMultinomial = "RemoveMultinomial"
+        ReportError = "ReportError"
+        AsOrdinal = "AsOrdinal"
+        AsNormalizedOrdinal = "AsNormalizedOrdinal"
+        Leave = "Leave"
```
Comment (kernc) [inline Orange/preprocess/preprocess.py]: I don't oppose strings at all. They are simple and robust. Sklearn is all strings in params, and nobody is complaining. But sklearn checks their user input for invalid values. Could we check the passed argument is valid? E.g.:
```py
assert multinomial_treatment in self._MultinomialTreatment.__dict__
```

File: Orange/preprocess/preprocess.py (hunk 1 above)
Comment (kernc) [inline Orange/preprocess/preprocess.py]: Can name this `MultinomialTreatment` is it's an enumeration of such methods?

**Comment by janezd** [comment]
Solved in #2409.

**Pull request closed without merging**